### PR TITLE
Add environmentInfo method to API

### LINKED ISSUE #183: Add environment/platform info API
As a user, I'd like to call an API to gather platform info, so I could use it to probe and aggregate all runtime details. 

This API, depending on the platform where it's invoked from, it could potentially give us these details: https://github.com/spring-cloud/spring-cloud-dataflow/wiki/Reporting-Issues

### PR DESCRIPTION
**Pull request by trisberg**
- add utility class to generate a base version for environment info

- add test to verify info contains required entries

Resolves #183

### CODE REVIEW

File: spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/AppDeployer.java [hunk 1]
```diff
@@ -135,4 +137,13 @@
...
 	AppStatus status(String id);
+
+	/**
+	 * Return the environment info for this deployer. This info should include the
+	 * following: 'deployer-class-name','deployer-version', 'deployer-spi-version', 'platform-type',
+	 * 'platform-client-version', 'platform-host-version', 'java-version', 'spring-version', 'spring-boot-version'.
+	 *
+	 * @return the app deployment environment info
+	 */
+	Map<String, String>  environmentInfo();
```
Comment (ericbottard) [inline spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/AppDeployer.java]: I'd +1.

Note that `deployer-classname` is something that 
1) is not very interesting to see for an end-user once you have the other properties
2) can be retrieved by doing `getClass()` on the very object implementing this interface

Comment (trisberg) [inline spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/AppDeployer.java]: OK, let me add a commit with a new return type, then we can argue what should be included.

I'll start with:

- deployerSpiVersion
- deployerImplementationVersion
- platformType
- platformClientVersion
- platformHostVersion
- javaVersion
- springVersion
- springBootVersion

**Comment by ghillert** [comment]
@trisberg I think we also need a Map<String,String> property in `DeployerEnvironmentInfo` for _optional_ or _platform specific_ properties. I would think that e.g. the CF implementation would want to add a slur of additional values.

**Comment by ghillert** [comment]
LGTM - merging.

**Comment by ghillert** [comment]
Merged with commit 076512c8aa90bfbc32e18034494564c06e5b595f

**Pull request closed without merging**